hydra-node/src/Hydra/HeadLogic/Outcome.hs

```diff
@@ -91,15 +91,20 @@ deriving anyclass instance (IsTx tx, ToJSON (ChainStateType tx)) => ToJSON (Stat
 deriving anyclass instance (IsTx tx, FromJSON (HeadState tx), FromJSON (ChainStateType tx)) => FromJSON (StateChanged tx)
 
 data Outcome tx
-  = Effects {effects :: [Effect tx]}
-  | StateChanged {event :: StateChanged tx}
-  | Wait {reason :: WaitReason tx}
-  | Error {error :: LogicError tx}
-  | Combined {left :: Outcome tx, right :: Outcome tx}
+  = -- | Continue with the given state updates and side effects.
+    Continue {events :: [StateChanged tx], effects :: [Effect tx]}
+  | -- | Wait for some condition to be met with optional state updates.
+    Wait {reason :: WaitReason tx, events :: [StateChanged tx]}
+  | -- | Processing resulted in an error.
+    Error {error :: LogicError tx}
   deriving stock (Generic)
 
 instance Semigroup (Outcome tx) where
-  (<>) = Combined
+  e@Error{} <> _ = e
+  _ <> e@Error{} = e
```

Comment:
I had an `error` here, which would bail on anyone trying to `mappend` anything onto it, but intuitively made this definition total.

To reviewers: should we make this partial to avoid shooting ourselves in the foot?

Reply:
Why would it be a problem to mappend here?

Reply:
Ah you refer to my comment: It's not a problem how this is now, just maybe unexpected as using `<>` on an `Error` and anything else would ignore the "anything". Given that we have quite big expressions, this might result in some gead scratches. Using `error` would fail directly

Reply:
I prefer not partial. We can always interrogate the value.

Reply:
I also lean to not use the error in the Monoid instance. Perhaps we can even remove the `Error` and instead use appropriate `ServerOutput` ?